libraries/AP_ExternalAHRS/AP_ExternalAHRS.cpp

```diff
@@ -271,7 +271,16 @@ bool AP_ExternalAHRS::pre_arm_check(char *failure_msg, uint8_t failure_msg_len)
         }
     }
 
-    if (!state.have_origin) {
+    bool check_origin = true;
+    // don't run this check for VN100 Series since its IMU only
+    if (DevType(devtype) == DevType::VecNav) {
```

Comment:
Instead of this VN-specific code, would it be reasonable to have something like 

```
if (num_gps_sensors() == 0) {
check_origin = false;
}
```

I'd generally prefer to avoid VN-specific code, and avoid parsing the string. This also leaves open the possibility later of adding a param to force you into "IMU only" mode, which I'd like to leave open